### PR TITLE
[tp] improve documentation

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115880

Improve the TP documentation in terms of format and descriptions

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

**Review by XilunWu — APPROVED** [review]
lgtm.

File: docs/source/distributed.tensor.parallel.rst [hunk 1]
```diff
@@ -32,7 +32,7 @@ Tensor Parallelism supports the following parallel styles:
 To simply configure the nn.Module's inputs and outputs with DTensor layouts
 and perform necessary layout redistributions, without distribute the module
 parameters to DTensors, the following classes can be used in
-the ``parallelize_plan``of ``parallelize_module``:
+the `parallelize_plan` of `parallelize_module`:
```
Comment (awgu) [inline docs/source/distributed.tensor.parallel.rst:35]: I thought that we need double backtick for code type and that single backtick is italics, but I may be wrong.

Comment (wanchaol) [reply docs/source/distributed.tensor.parallel.rst:35]: It's kind of weird to me too, but the lastest main docs looks wrong format so I switched back to single backtick, see lines before this API https://pytorch.org/docs/main/distributed.tensor.parallel.html#torch.distributed.tensor.parallel.PrepareModuleInput

I'll play around with the code types a bit more and see the refreshed docs before landing

Comment (awgu) [reply docs/source/distributed.tensor.parallel.rst:35]: Maybe it is missing the space between the last backtick of \`\`parallelize_plan\`\` and of, causing it to render improperly?

Comment (wanchaol) [reply docs/source/distributed.tensor.parallel.rst:35]: good point, lm try that instead

Comment (wanchaol) [reply docs/source/distributed.tensor.parallel.rst]: adjusted multiple places to double backtick , but the built doc still rendering the old version, gonna land this first and see the doc

**Pull request closed without merging**